Bottom: facc0b7417d101a364ef97d4c9bbae5942e7e109
Top:    3afb51c6c6d08a52ef72ca238aa3e1406caad752
Author: Ian Kent <raven@themaw.net>
Date:   2013-08-19 12:45:52 +0800

Refresh of autofs-5.0.7-allow-use-of-hosts-map-in-maps.patch

---

diff --git a/modules/parse_sun.c b/modules/parse_sun.c
index 9a12028..86e3f48 100644
--- a/modules/parse_sun.c
+++ b/modules/parse_sun.c
@@ -1595,13 +1595,17 @@ int parse_mount(struct autofs_point *ap, const char *name,
 			p = skipspace(p);
 		}
 
-		loclen = strlen(loc);
-		if (loclen == 0) {
-			free(loc);
-			free(options);
-			error(ap->logopt,
-			      MODPREFIX "entry %s is empty!", name);
-			return 1;
+		/* if it's not a hosts map loc must be non-null */
+		if (!(strstr(options, "fstype=autofs") &&
+		      strstr(options, "hosts"))) {
+			loclen = strlen(loc);
+			if (loclen == 0) {
+				free(loc);
+				free(options);
+				error(ap->logopt,
+				      MODPREFIX "entry %s is empty!", name);
+				return 1;
+			}
 		}
 
 		debug(ap->logopt,
